### PR TITLE
Invalid Mouse Over events on NC raw events

### LINKED ISSUE #7748: Tooltip appears when resizing
**Describe the bug**
When resizing action starts, current mouse position is remembered and later tested against hit testing, even during resizing. That leads to buttons getting focused/unfocused, tooltips being displayed/hidden. Furthermore, when a tooltip appears and I try to keep resizing, applications stops resizing actions and reverts size back to starting value.

**To Reproduce**

1. Open attached solution 
[AvaloniaResizingTooltip.zip](https://github.com/AvaloniaUI/Avalonia/files/8185885/AvaloniaResizingTooltip.zip)
2. Start resizing bottom left corner
3. Resize it in a way that a buttons moves over resize start position
4. Button gets focused and its tooltip appears
5. Continue resizing, preferably in a circular movement
6. Resizing is stopped unexpectedly 

**Expected behavior**
Button doesn't get focused nor its tooltip appears during resize action.

**Screenshots**
ResizeToolTipBug

**Desktop (please complete the following information):**

- OS: Both PC and Mac

### PR DESCRIPTION
**Pull request by ltetak**
## What does the pull request do?
When the user resizes a window it looks like we get a mouseover on a weird window's positions. This can result in the user acting on a window (e.g. activating tooltips, flyouts, etc.) which is disturbing. 
We shouldn't notify the window about mouse events when he/she is resizing the window.

## What is the current behavior?
When you resize a window you will get all kinds of mouse events as if the user was moving the mouse over the window (but in the wrong position). 

## What is the updated/expected behavior with this PR?
Users will get mouse move notifications only if they really move the mouse above the window.

## How was the solution implemented (if it's not obvious)?
Code around mouse move events is not entirely consolidated. On this line
https://github.com/AvaloniaUI/Avalonia/compare/master...ltetak:feature/mouse_over_events?expand=1#diff-ac4f971803c43a9eb47a48db84d4c676d2678cb67d7cb60303b888cf49af4244R152
we remember the position of a mouse event in a class field and this value is then used on SceneInvalidated event. I don't understand the full reasoning but this fails if the position is not correct. I think the only situation is when the raw event is NonClient event (tested on both PC and Mac). The mouse position is still the old one. We can safely ignore it in this case.

## Breaking changes
I don't think anybody could rely on these events because they are not distinguishable from other pointer-events and the position is clearly wrong.

## Fixed issues
Fixes: https://github.com/AvaloniaUI/Avalonia/issues/7748

### CODE REVIEW

File: src/Avalonia.Input/MouseDevice.cs [hunk 1]
```diff
@@ -147,6 +147,8 @@ private void ProcessRawEvent(RawPointerEventArgs e)
             if(mouse._disposed)
                 return;
 
+            if (e.Type == RawPointerEventType.NonClientLeftButtonDown) return;
+
             _position = e.Root.PointToScreen(e.Position);
```
Comment (ltetak) [inline src/Avalonia.Input/MouseDevice.cs:152]: This position is stored and used as a parameter for SetPointerOver calls here
https://github.com/AvaloniaUI/Avalonia/blob/ee648c7c2c76cb9bcdf3b0833890aeafe30ea083/src/Avalonia.Input/MouseDevice.cs#L109-L123
This position is not correct for `RawPointerEventType.NonClientLeftButtonDown`-s though.

Comment (MarchingCube) [reply src/Avalonia.Input/MouseDevice.cs:152]: Position **was** correct at click time but due to resizing it cached a stale value. But anyway this old logic was conflicting with window pointer leave mechanic (which was clearing pointerover state) so your fix looks fine to me.

**Review by MarchingCube — APPROVED** [review]
Looks great, this is resolving a long time bug that I was seeing my apps that I never had time to investigate. And all it took was a one liner :)

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019296-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**